### DISCUSSION TITLE
Nordic NRF52840: Bluetooth Central HIDS Error

### DISCUSSION THREAD

**Discussion by 2songh** [zephyrproject-rtos/zephyr#69779] [Q&A]
Zephyr Version: 3.4.99
Board: nrf52840dk_nrf52840

Use Sample: zephyrproject/zephyr/samples/bluetooth/encrypted_advertising/central

I would like to use the NRF52840 board as a central device to connect a bluetooth HID keyboard to it.

**procedure**
1. west build -b nrf52840dk_nrf52840
2. west flash
3. NRF52840 Booting
4. Start scanning for nearby Bluetooth devices
5. Bluetooth keyboard advertising starts
6. Connecting boards and keyboard
7. Check passkey
![image](https://github.com/zephyrproject-rtos/zephyr/assets/126455296/42817fc7-d0fb-4ccd-9125-332903b31b83)
8. Click Button1 on the NRF52840 board
9. Security Change
![image](https://github.com/zephyrproject-rtos/zephyr/assets/126455296/ae4c03b0-0225-4042-9846-79c6c036afbf)
10. LED output from my Bluetooth keyboard to indicate normal connection
11. Error message output from NRF52840  board
![image](https://github.com/zephyrproject-rtos/zephyr/assets/126455296/aec3893e-88c0-404a-8beb-e6f617ecb067)

Errors occur in gatt_read() & gatt_read_cb().

Is it a GATT settings issue?
Or is there a problem with HID service function support?

Please give me advice regarding Bluetooth HID.

**Comment by hermabe** [comment]
I am not sure what you are asking here. The Encrypted Advertising Data (EAD) sample has a central and a peripheral part and they are made to be used together. They will not work with a different sample that is not made to use EAD. As far as I know does BLE HID not use EAD.

The error code is `BT_ATT_ERR_INVALID_HANDLE` which is probably caused by the GATT server (on the peripheral) not having the custom GATT service with key material.

**Reply by 2songh** [reply]
Sorry for the lack of explanation.

I modified the unnecessary parts of the EAD sample and built it.
I'm curious about how to configure central hid.

Is there a sample related to this?

**Reply by hermabe** [reply]
If you are using nordic nrfconnect (which I assume since there is no central HIDS sample in upstream zephyr), you can use the `peripheral_hids_keyboard` sample together with the `central_hids` sample as described in the docs: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/bluetooth/central_hids/README.html#testing

**Reply by 2songh** [reply]
Thank you for your advice.
I will leave an additional answer after testing.

**Reply by 2songh** [reply]
I have additional questions.

Isn't EAD only used for the Bluetooth advertising function?
Is EAD also used when reading Bluetooth keyboard data?

I'm wondering if it's possible to only use EAD for Bluetooth advertising, and then receive HID data afterwards.

**Reply by hermabe** [reply]
EAD is encrypted advertising data and is as such only used for advertising. HID reports are sent over a connection which can be both encrypted and unencrypted.

**Reply by 2songh** [reply]
Create a connected state after advertising,
Need to connect and communicate with HID reports?

**Reply by hermabe** [reply]
HID reports are sent over connections yes. Please read the documentation for the samples (eg. https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/bluetooth/central_hids/README.html).

If you are not familiar with how Bluetooth LE works I suggest you read up on it.

**Reply by 2songh** [reply]
Thank you for your advice.
I will conduct additional test.
